worlds/ff4fe/__init__.py

```diff
@@ -0,0 +1,488 @@
+        placement_dict["seed"] = self.player + self.multiworld.seed
+        placement_dict["output_file"] = f'{self.multiworld.get_out_file_name_base(self.player)}' + '.sfc'
+        placement_dict["flags"] = flags.create_flags_from_options(self.options, self.objective_count)
+        placement_dict["junk_tier"] = self.options.JunkTier.value
+
+        # Our actual patch is just a set of instructions and data for FE to use.
+        patch = FF4FEProcedurePatch(player=self.player, player_name=self.multiworld.player_name[self.player])
+        patch.write_file("placement_file.json" , json.dumps(placement_dict).encode("UTF-8"))
+        rom_path = os.path.join(
+            output_directory, f"{self.multiworld.get_out_file_name_base(self.player)}" f"{patch.patch_file_ending}"
+        )
+        patch.write(rom_path)
+
+    def fill_slot_data(self) -> Mapping[str, Any]:
+        # Slot data needed for tracker logic.
+        slot_data = {
+            "DarkMatterHunt": self.options.FindTheDarkMatter.current_key,
+            "NoEarnedCharacters": self.options.NoEarnedCharacters.current_key,
+            "NoFreeCharacters": self.options.NoEarnedCharacters.current_key,
+            "PassEnabled": self.options.PassEnabled.current_key,
+            "AdditionalObjectives": self.options.AdditionalObjectives.value,
+            "ObjectiveReward": self.options.ObjectiveReward.current_key,
+            "UnsafeKeyItemPlacement": self.options.UnsafeKeyItemPlacement.current_key,
+            "ObjectivesRequired": self.options.RequiredObjectiveCount.value
+        }
+        return slot_data
+
+    def create_placement_file(self, rom_name):
+        placement_dict = {"rom_name": rom_name}
+        for location in self.multiworld.get_filled_locations(self.player):
+            # Placement dictionary doesn't need AP event logic stuff.
+            if location.name in [event.name for event in events.boss_events] or location.name.startswith("Objective"):
+                continue
+            location_data = [loc for loc in all_locations if loc.name == location.name].pop()
+            if location.item.player == self.player:
+                item_data = [item for item in all_items if item.name == location.item.name].pop()
+                placement_dict[location_data.fe_id] = {
+                    "location_data": location_data.to_json(),
+                    "item_data": item_data.to_json(),
+                    "item_name": item_data.name,
+                    "player_name": self.multiworld.player_name[location.item.player]
+                }
+            else:
+                placement_dict[location_data.fe_id] = {
+                    "location_data": location_data.to_json(),
+                    "item_data": ItemData.create_ap_item().to_json(),
+                    "item_name": location.item.name,
+                    "player_name": self.multiworld.player_name[location.item.player]
```

Comment:
```suggestion
                    "player_name": self.player_name,
```

Reply:
The placement dictionary includes names for all player's items, not just the FE player. I need to use the lookup to get those names too so text like "got Silver Sword for Player5" can be written into the game.